EchoPro/survey.py

```diff
@@ -363,6 +364,155 @@ def biometric_distributions( self ):
         # Push to biology attribute 
         self.biology['distributions'] = biometrics
 
+    @staticmethod
+    def transect_analysis():
+        # INPUTS
+        # This is where the users can designate specific transect numbers,
+        # stratum numbers, species, etc. These would be applied to the functions
+        # below
+        
+        # Initialize new attribute
+        self.results = {}
+        #### TODO: THIS SHOULD BE ADDED TO THE ORIGINAL SURVEY OBJECT CREATION
+        #### THIS IS INCLUDED HERE FOR NOW FOR TESTING PURPOSES -- ALL CAPS IS CRUISE
+        #### CONTROL FOR "REMEMBER TO MAKE THIS CHANGE BRANDYN !!!"
+                
+        # Calculate sigma_bs per stratum        
+        self.strata_mean_sigma_bs()
+        # OUTPUT: self.acoustics['sigma_df'] (pd.DataFrame)
+        
+        # Fill in missing strata data 
+        ### TODO: This does not necessarily need to be its own function
+        ### It also does not need to resemble the original source code (ie for loop)
+        missing_strata = []
+        for stratum in self.spatial['strata_df']['stratum_num']:
+            if stratum not in self.acoustics['sigma_df']['stratum_num']:
+                missing_strata.append(stratum)
+        # OR 
+        self.impute_missing_strata()
+        
+        # Fill in missing sigma_bs values
+        self.impute_missing_sigma_bs()
```

Comment:
I think that is certainly workable. Would it be too much to embed both `impute_missing_strata()` and `impute_missing_sigma_bs()` within `strata_mean_sigma_bs()`? I can see how that may obfuscate the fact that the underlying (modeled) data are being imputed and therefore be a tad more difficult to debug/parse.

Reply:
I agree that it would be ok to embed both  `impute_missing_strata()` and `impute_missing_sigma_bs()` within `strata_mean_sigma_bs()`, so that `strata_mean_sigma_bs` will output `sigma_bs` for each strata, regardless of what happens under the hood. The imputation seems a bit too detailed to split out, but is something that could/should be explained in the docstring, so that users are reminded that this may have happened to their data.

Actually, how about if you indeed embed everything under `strata_mean_sigma_bs` and print out messages like "Length data missing for stratum A, using strata B-C for estimating sigma_bs for stratum A" for each of the missing strata?